### ISSUE TITLE
CommonJS modules

### ISSUE THREAD

**Issue by inca** [Voog/wysihtml#150]

**Comment by inca** [comment]
Looking forward for core team opinion on this one. If you don't mind the big changes, I can assist in rewriting modules into CommonJS format.

**Comment by pulges** [comment]
Current plan is to make it RequireJS/AMD compliant while keeping the option to run it as standalone without requireJS. Also the ability for users to extend the module with custom commands etc should be kept. No analysis about how much refactoring and api changes it will take, but the changes have to be done eventually anyway.  It is currently not in our short-list, but all external help is welcome and handled ASAP.

**Comment by inca** [comment]
Thanks for your feedback! My opinion is pretty inline with [this one](http://dontkry.com/posts/code/browserify-and-the-universal-module-definition.html). In my practice module loader sniffing is becoming pretty annoying now that people tend to assemble their JavaScript assets using server-side tooling like [Browserify](http://browserify.org/) and [Uglify](https://github.com/mishoo/UglifyJS). Modules which try to be "too smart" (UMD) always require special treatment (with derequire, deamdify, derequire, debowerify, decomponentify, etc.).

Now, I've already scouted through sources and sketched up a plan for migration. It would, unsurprisingly, affect every js file out there; however, some additional reorganization required to eliminate cyclic dependencies (since right now most modules are bundled onto the `wysihtml5` namespace, while using constants defined on `wysihtml5` object at the same time). Introducing separate `constants` module would resolve this issue, while leaving `wysihtml5` as top-level exports object.

To sum things up: it is kinda doable, but would require some effort.

**Comment by pulges** [comment]
The new version 0.6.0-beta has removed the bundled rangy UMD wrappers now. Also does not add any other UMD wrappers itself, to comply better with server side bundlers and leaving open the door for ES6 module pattern if it starts working in browser. Any other blocking problems when bundling or can the issue be closed?

**Comment by trickeyone** [comment]
I anyone else comes across this, I did go through and make it AMD/CommonJS compliant. It may be a little dated, though. I made it about 6 months ago.
https://github.com/trickeyone/wysihtml
